controller/konnect/ops/ops_konnectdataplanegroupconfigurations.go

```diff
@@ -0,0 +1,247 @@
+			n.Status.DataPlaneGroups = dataPlaneGroupsResponseToStatus(resp.ConfigurationManifest.GetDataplaneGroups())
+			return nil
+		}
+
+		// If there was an error which wasn't a conflict, complaining about submitting
+		// the same configuration, wrap it and return it.
+		if errWrap := wrapErrIfKonnectOpFailed(err, CreateOp, n); errWrap != nil {
+			return errWrap
+		}
+	}
+
+	if resp == nil || resp.ConfigurationManifest == nil || resp.ConfigurationManifest.ID == "" {
+		return fmt.Errorf("failed updating %s: %w", n.GetTypeName(), ErrNilResponse)
+	}
+
+	// At this point, the DataPlaneGroupConfiguration has been created in Konnect.
+	id := resp.ConfigurationManifest.ID
+	n.SetKonnectID(id)
+	n.Status.DataPlaneGroups = dataPlaneGroupsResponseToStatus(resp.ConfigurationManifest.GetDataplaneGroups())
+
+	// sdk.ListConfigurations(ctx, sdkkonnectops.ListConfigurationsRequest{
+	// 	Filter: &sdkkonnectcomp.ConfigurationsFilterParameters{
+	// 		ControlPlaneID:  *sdkkonnectcomp.IDFieldFilter,
+	// 		ControlPlaneGeo: *sdkkonnectcomp.ControlPlaneGeoFieldFilter,
+	// 	},
+	// })
+
+	return nil
+}
+
+// deleteKonnectDataPlaneGroupConfiguration deletes a Konnect DataPlaneGroupConfiguration.
+// It is assumed that the Konnect DataPlaneGroupConfiguration has a Konnect ID.
+func deleteKonnectDataPlaneGroupConfiguration(
+	ctx context.Context,
+	sdk sdkops.CloudGatewaysSDK,
+	n *konnectv1alpha1.KonnectCloudGatewayDataPlaneGroupConfiguration,
+) error {
+	cpID := n.GetControlPlaneID()
+	if cpID == "" {
+		return CantPerformOperationWithoutControlPlaneIDError{Entity: n, Op: DeleteOp}
+	}
+	// NOTE: we delete the data plane group configuration by "creating" (using PUT)
+	// a new configuration with the same ID and the same version, but with an empty
+	// list of data plane groups.
+	req := cloudGatewayDataPlaneGroupConfigurationInit(n.Spec, cpID)
+	resp, err := sdk.CreateConfiguration(ctx, req)
+
+	if errWrap := wrapErrIfKonnectOpFailed(err, CreateOp, n); errWrap != nil {
+		return errWrap
+	}
+
+	if resp == nil || resp.ConfigurationManifest == nil || resp.ConfigurationManifest.ID == "" {
+		return fmt.Errorf("failed deleting %s: %w", n.GetTypeName(), ErrNilResponse)
+	}
+
+	return nil
+}
+
+func cloudGatewayDataPlaneGroupConfigurationInit(
+	spec konnectv1alpha1.KonnectCloudGatewayDataPlaneGroupConfigurationSpec,
+	cpID string,
+) sdkkonnectcomp.CreateConfigurationRequest {
+	return sdkkonnectcomp.CreateConfigurationRequest{
+		ControlPlaneID: cpID,
+		Version:        spec.Version,
+		APIAccess:      spec.APIAccess,
+		// TODO deduct CP geo
+		ControlPlaneGeo: sdkkonnectcomp.ControlPlaneGeoEu,
+		DataplaneGroups: []sdkkonnectcomp.CreateConfigurationDataPlaneGroup{},
+	}
+}
+
+func cloudGatewayDataPlaneGroupConfigurationToAPIRequest(
+	spec konnectv1alpha1.KonnectCloudGatewayDataPlaneGroupConfigurationSpec,
+	cpID string,
+) sdkkonnectcomp.CreateConfigurationRequest {
+	ret := cloudGatewayDataPlaneGroupConfigurationInit(spec, cpID)
```

Comment:
I would rather rename this `ret` to something like `cfgReq` to better correspond with return value type

Reply:
9573e176ae24fce02d506e18e15c8f5312256ae2